components/chainhook-cli/src/scan/stacks.rs

```diff
@@ -338,7 +340,12 @@ pub async fn scan_stacks_chainstate_via_rocksdb_using_predicate(
             apply: hits_per_blocks,
             rollback: vec![],
         };
-        let res = match handle_stacks_hook_action(trigger, &proofs, ctx) {
+        let res = match handle_stacks_hook_action(
+            trigger,
+            &proofs,
+            &config.get_event_observer_config().predicates_config,
```

Comment:
It seems more reasonable to pass the parent `EventObserverConfig` here and let the `handle_stacks_hook_action` function decide what it needs. The caller don't need to know about implementation details of the callee function.